src/parser/jobs/mnk/GreasedLightning.js

```diff
@@ -0,0 +1,305 @@
+	}
+
+	_onComplete() {
+		// Push the final GL count so that it lasts to the end of the fight
+		this._stacks.push({stacks: 0, timestamp: this.parser.fight.end_time})
+
+		// Push wasted saves for failed RoE
+		this._earthSaves.forEach(earth => {
+			if (!earth.clean) {
+				this._wastedSaves++
+			}
+		})
+
+		this.checklist.add(new Rule({
+			name: 'Keep Greased Lightning running',
+			description: <Fragment>
+				<StatusLink {...STATUSES.GREASED_LIGHTNING_I}/> is a huge chunk of MNK's damage, increasing your damage by 30% and attack speed by 15%.
+			</Fragment>,
+			requirements: [
+				new Requirement({
+					name: <Fragment><StatusLink {...STATUSES.GREASED_LIGHTNING_I}/> uptime</Fragment>,
+					percent: () => this.getUptimePercent(),
+				}),
+			],
+			// Assuming slowest possible GCD, using 1 TK every 90s should be just over 92% uptime
+			// TODO: use a metric based on good TK recovery to adjust this lower
+			target: 92,
+		}))
+
+		if (this._droppedStacks > 0) {
+			this.suggestions.add(new Suggestion({
+				icon: ACTIONS.TORNADO_KICK.icon,
+				content: <Fragment>
+					Avoid dropping stacks except when using <ActionLink {...ACTIONS.TORNADO_KICK} />.
+				</Fragment>,
+				severity: SEVERITY.MAJOR,
+				why: <Fragment>
+					<StatusLink {...STATUSES.GREASED_LIGHTNING_I} /> dropped {this._droppedStacks} times.
+				</Fragment>,
+			}))
+		}
+
+		if (this._wastedSaves > 0) {
+			this.suggestions.add(new Suggestion({
+				icon: ACTIONS.RIDDLE_OF_EARTH.icon,
+				content: <Fragment>
+					Check the fight timeline to see when you can save <StatusLink {...STATUSES.GREASED_LIGHTNING_I} /> with <ActionLink {...ACTIONS.RIDDLE_OF_EARTH} />.
+				</Fragment>,
+				severity: SEVERITY.MINOR,
+				why: <Fragment>
+					<ActionLink {...ACTIONS.RIDDLE_OF_EARTH} /> was used {this._wastedSaves} times without preserving <StatusLink {...STATUSES.GREASED_LIGHTNING_I} />,
+				</Fragment>,
+			}))
+		}
+	}
+
+	getUptimePercent() {
+		const fightUptime = this.parser.fightDuration - this.invuln.getInvulnerableUptime()
+
+		const statusUptime = this._stacks.reduce((duration, value, index) => {
+			const last = this._stacks[index-1] || {}
+			if (value.stacks === 0 && last.stacks === GL_MAX_STACKS) {
+				duration += value.timestamp - last.timestamp
+			}
+
+			return duration
+		}, 0)
+
+		return (statusUptime / fightUptime) * 100
+	}
+
+	output() {
+		// TODO: figure out how to make this graph at least 3x shorter in height
+
+		// Disabling magic numbers for the chart, 'cus it's a chart
+		/* eslint-disable no-magic-numbers */
+		const data = {
+			datasets: [{
+				label: 'GL Stacks',
+				data: this._stacks.map(({stacks, timestamp}) => ({y: stacks, t: timestamp - this.parser.fight.start_time})),
+				backgroundColor: Color(JOBS.MONK.colour).fade(0.5),
+				borderColor: Color(JOBS.MONK.colour).fade(0.2),
+				steppedLine: true,
+			}],
+		}
+
+		const options = {
+			legend: {display: false},
+			tooltips: {enabled: false},
+			scales: {
+				xAxes: [{
```

Comment:
With the tweaks to the core chart component, you can probably completely remove most of these options. Anything defined in the component's default options will be set unless explicitly overwritten. :blobmorning: